### DISCUSSION TITLE
Unexpected Ex field in basic Laser Example 2D

### DISCUSSION THREAD

**Discussion by erny123** [epochpic/epoch#144] [Q&A] [answered]
Dear EPOCH developers,

I am running the basic 2D Laser example on the latest EPOCH-14.17-14 and it is showing an electric field in the x-direction (Ex) to be of 1 order of magnitude smaller with a weird profile around the axis. I noticed this problem in an earlier version of 14.17.10 also.
And there doesn't seem to be any difference with using CPML boundaries. 

Edit: Looks to me like it might be some unexpected longitudinal modes corresponding to 1,0 or 0,1 modes. 
What's interesting is that the magnetic fields stay valid aren't developing longitudinal modes, which might be expected even with longitudinal E-field?

![image](https://user-images.githubusercontent.com/34816295/108009563-db11b080-6fb7-11eb-9386-cc03241d3aba.png)
![image](https://user-images.githubusercontent.com/34816295/108009577-e1a02800-6fb7-11eb-9f14-17c6fd5bc8fb.png)
![image](https://user-images.githubusercontent.com/34816295/108010813-9dfaed80-6fba-11eb-983b-14804d1fd906.png)

```
begin:control
  nx = 500
  ny = nx

  # Size of domain
  x_min = -10 * micron
  x_max = -x_min
  y_min = x_min
  y_max = x_max

  # Final time of simulation
  t_end = 50 * femto

  stdout_frequency = 10
end:control


begin:boundaries
  bc_x_min = simple_laser
  bc_x_max = open
  bc_y_min = periodic
  bc_y_max = periodic
end:boundaries


begin:constant
  lambda0 = 1 * micron
  theta = pi / 8.0
end:constant


begin:laser
  boundary = x_min
  intensity_w_cm2 = 1.0e15
  lambda = lambda0 * cos(theta)
  profile = gauss(y, 0, 4*micron)
  #phase = -2.0 * pi * y * tan(theta) / lambda0
  #t_profile = gauss(time, 2*micron/c, 1*micron/c)
end:laser


begin:output
  dt_snapshot = 1 * micron / c

  # Properties on grid
  grid = always
  ex = always
  ey = always
  ez = always
  bx = always
  by = always
  bz = always
end:output

```

**Comment by keithbennett** [comment, marked as answer]
This looks like the correct solution to me. In a plane-wave setup, there is no structure in the y-direction and the only EM components will be Ey and Bz. However, the Gaussian profile has a gradient in the y-direction and therefore an Ex component will be generated via the equation

![\frac{d B_z}{dy} = \mu_0 J_x + \frac{1}{c^2}\frac{d E_x}{dt}](https://render.githubusercontent.com/render/math?math=%5Cdisplaystyle+%5Cfrac%7Bd+B_z%7D%7Bdy%7D+%3D+%5Cmu_0+J_x+%2B+%5Cfrac%7B1%7D%7Bc%5E2%7D%5Cfrac%7Bd+E_x%7D%7Bdt%7D)

The gradient of Bz switches sign at y=0, so this will corresponds to the Ex profile observed.

**Reply by erny123** [reply]
Wow, yes that makes perfect sense. Thanks for the clarification.
